Mds/Collivery.php

```diff
@@ -98,850 +66,887 @@ protected function client()
-			return $this->cache->get( 'collivery.parcel_image_list.'. $this->client_id .'.'. $collivery_id );
-		} else {
-			try {
-				$result = $this->client()->get_parcel_image_list( $collivery_id, $this->token );
-			} catch ( SoapFault $e ) {
-				$this->catchSoapFault( $e );
-				return false;
-			}
-
-			if ( isset( $result['images'] ) ) {
-				if ( isset( $result['error_id'] ) )
-					$this->setError( $result['error_id'], $result['error'] );
-				else
-					if ( $this->check_cache != 0 ) $this->cache->put( 'collivery.parcel_image_list.'. $this->client_id .'.'. $collivery_id, $result['images'], 60*12 );
-
-					return $result['images'];
-			} else {
-				if ( isset( $result['error_id'] ) )
-					$this->setError( $result['error_id'], $result['error'] );
-				else
-					$this->setError( 'result_unexpected', 'No result returned.' );
-
-				return false;
-			}
-		}
-	}
-
-	/**
-	 * Returns the image of a given parcel-id of a waybill.
-	 * If the Waybill number is 54321 and there are 3 parcels, they would
-	 * be referenced by id's 54321-1, 54321-2 and 54321-3.
-	 *
-	 * @param string  $parcel_id Parcel ID
-	 * @return array               Array containing all the information
-	 *                             about the image including the image
-	 *                             itself in base64
-	 */
-	public function getParcelImage( $parcel_id )
-	{
-		if ( ( $this->check_cache == 2 ) && $this->cache->has( 'collivery.parcel_image.'. $this->client_id .'.'. $parcel_id ) ) {
-			return $this->cache->get( 'collivery.parcel_image.'. $this->client_id .'.'. $parcel_id );
-		} else {
-			try {
-				$result = $this->client()->get_parcel_image( $parcel_id, $this->token );
-			} catch ( SoapFault $e ) {
-				$this->catchSoapFault( $e );
-				return false;
-			}
-
-			if ( isset( $result['image'] ) ) {
-				if ( isset( $result['error_id'] ) )
-					$this->setError( $result['error_id'], $result['error'] );
-				else
-					if ( $this->check_cache != 0 ) $this->cache->put( 'collivery.parcel_image.'. $this->client_id .'.'. $parcel_id, $result['image'], 60*24 );
-
-					return $result['image'];
-			} else {
-				if ( isset( $result['error_id'] ) )
-					$this->setError( $result['error_id'], $result['error'] );
-				else
-					$this->setError( 'result_unexpected', 'No result returned.' );
-
-				return false;
-			}
-		}
+     * Make the authentication request.
+     *
+     * @param null|array $settings
+     *
+     * @return array
+     */
+    public function makeAuthenticationRequest($settings = null)
+    {
+        if ($settings) {
+            $user_email = $settings['email'];
+            $user_password = $settings['password'];
+            $token = null;
+        } else {
+            $user_email = $this->config->user_email;
+            $user_password = $this->config->user_password;
+            $token = $this->token;
+        }
+
+        try {
+
+            $authenticate = $this->consumeAPI('https://api.collivery.co.za/v3/login', [
```

Comment:
Please don't pass in full url into `$this->consumeApi()` 
Please refactor so that the base `https://api.collivery.co.za/v3` is not repeated in all calls